qiskit/circuit/library/standard_gates/multi_control_rotation_gates.py

```diff
@@ -230,13 +337,11 @@ def mcrz(
     self._check_dups(all_qubits)
 
     n_c = len(control_qubits)
-    if n_c == 1:  # cu
-        _apply_cu(self, 0, 0, lam, control_qubits[0], target_qubit, use_basis_gates=use_basis_gates)
+    if n_c == 1:
+        self.compose(CRZGate(lam), control_qubits + [target_qubit], inplace=True)
```

Comment:
I think `append` would be the more adequate function to use here, as it's specifically for instructions/gates 🙂 
```suggestion
        self.append(CRZGate(lam), control_qubits + [target_qubit])
```